module/designer/migrations/Version20210125121500.php

```diff
@@ -0,0 +1,104 @@
+            FROM designer.template_element WHERE properties->\'attribute_id\' iS NOT NULL
+            )
+        ');
+
+        $this->addSql('UPDATE designer.template_element SET type = properties->>\'variant\'');
+
+        $this->addSql('ALTER TABLE designer.template_element ALTER COLUMN type DROP DEFAULT');
+        $this->addSql('ALTER TABLE designer.template_element ALTER COLUMN type SET NOT NULL');
+
+        $this->addSql('ALTER TABLE designer.template_element DROP COLUMN properties');
+
+        $this->addSql('
+            UPDATE event_store 
+            SET payload = jsonb_set(payload, \'{element, type}\', payload->\'element\'->\'properties\'->\'variant\') 
+            WHERE event_id IN (
+                SELECT id 
+                FROM event_store_event 
+                WHERE event_class = \'Ergonode\Designer\Domain\Event\TemplateElementAddedEvent\'
+                OR event_class =  \'Ergonode\Designer\Domain\Event\TemplateElementChangedEvent\'
+            )');
+
+        $this->addSql('
+            UPDATE event_store 
+            SET payload = jsonb_insert(payload,\'{element, attribute_id}\', 
+                payload->\'element\'->\'properties\'->\'attribute_id\')
+            WHERE event_id IN (
+                SELECT id 
+                FROM event_store_event 
+                WHERE event_class = \'Ergonode\Designer\Domain\Event\TemplateElementAddedEvent\'
+                OR event_class =  \'Ergonode\Designer\Domain\Event\TemplateElementChangedEvent\'
+                )
+            AND payload->\'element\'->\'properties\'->\'attribute_id\' IS NOT NULL');
+
+        $this->addSql('
+            UPDATE event_store 
+            SET payload = jsonb_insert(payload,\'{element, required}\', 
+                payload->\'element\'->\'properties\'->\'required\')
+            WHERE event_id IN (
+                SELECT id 
+                FROM event_store_event 
+                WHERE event_class = \'Ergonode\Designer\Domain\Event\TemplateElementAddedEvent\'
+                OR event_class =  \'Ergonode\Designer\Domain\Event\TemplateElementChangedEvent\'
+                )
+            AND payload->\'element\'->\'properties\'->\'required\' IS NOT NULL');
+
+        $this->addSql('
+            UPDATE event_store 
+            SET payload = jsonb_insert(payload,\'{element, label}\', payload->\'element\'->\'properties\'->\'label\')
+            WHERE event_id IN (
+                SELECT id 
+                FROM event_store_event 
+                WHERE event_class = \'Ergonode\Designer\Domain\Event\TemplateElementAddedEvent\'
+                OR event_class =  \'Ergonode\Designer\Domain\Event\TemplateElementChangedEvent\'
+                )
+            AND payload->\'element\'->\'properties\'->\'label\' IS NOT NULL');
+
+        $this->addSql('
+            UPDATE event_store 
+            SET payload  = payload #- \'{element,properties}\' 
```

Comment:
this comment in sql is ok ?

Reply:
sql use  -- not # this is json operator
https://www.postgresql.org/docs/9.5/functions-json.html